config/listenaddress/listenaddress.go

```diff
@@ -42,59 +45,60 @@ const (
 
 // Configuration is the configuration for resolving a listen address.
 type Configuration struct {
-	// Hostname is the hostname to use
-	Hostname string `yaml:"hostname" validate:"nonzero"`
-
-	// Port specifies the port to listen on
-	Port *Port `yaml:"port" validate:"nonzero"`
-}
+	// ListenAddressType is the port type for the port
+	ListenAddressType Resolver `yaml:"type" validate:"nonzero"`
 
-// Port specifies the port to use
-type Port struct {
-	// PortType is the port type for the port
-	PortType Resolver `yaml:"portType" validate:"nonzero"`
+	// Value is the config specified listen address if using config port type.
+	Value *string `yaml:"value"`
 
-	// Value is the config specified port if using config port type.
-	Value *int `yaml:"value"`
+	// EnvVarListenPort specifies the environment variable name for the listen address port.
+	EnvVarListenPort *string `yaml:"envVarListenPort"`
 
-	// EnvVarName is the environment specified port if using environment port type.
-	EnvVarName *string `yaml:"envVarName"`
+	// EnvVarListenHost specifies the environment variable name for the listen address hostname.
+	EnvVarListenHost *string `yaml:"envVarListenHost"`
 }
 
 // Resolve returns the resolved listen address given the configuration.
 func (c Configuration) Resolve() (string, error) {
-	if c.Port == nil {
-		return "", errors.New("missing port config")
-	}
-	p := c.Port
 
-	var port int
-	switch p.PortType {
+	portType := c.ListenAddressType
```

Comment:
nit: Change `portType` local var to `addressType` or `listenAddrType` or similar which accurately reflects the variable value?